alacritty/src/input/keyboard.rs

```diff
@@ -218,7 +217,30 @@ impl<T: EventListener, A: ActionContext<T>> Processor<T, A> {
                 *suppress_chars.get_or_insert(true) &= binding.action != Action::ReceiveChar;
 
                 // Binding was triggered; run the action.
-                binding.action.clone().execute(&mut self.ctx);
+                Some(binding.action.clone())
+            } else {
+                None
+            }
+        };
+
+        // Trigger matching key bindings.
+        for i in 0..self.ctx.config().key_bindings().len() {
+            let binding = &self.ctx.config().key_bindings()[i];
+            if let Some(action) = binding_action(binding) {
+                action.execute(&mut self.ctx);
+            }
+        }
+
+        // Trigger key bindings for hints.
+        for i in 0..self.ctx.config().hints.enabled.len() {
+            let hint = &self.ctx.config().hints.enabled[i];
+            let binding = match hint.binding.as_ref() {
+                Some(binding) => binding.key_binding(&hint),
+                None => continue,
+            };
+
+            if let Some(action) = binding_action(binding) {
+                action.execute(&mut self.ctx);
             }
```

Comment:
You could incorporate the conditional and execute parts into the helper closure I believe.

Reply:
No, because that would pass a reference (binding) and mutable reference (self.ctx).